packages/store-cache/src/createDatabaseClient.ts

```diff
@@ -10,28 +10,40 @@ import {
 } from "./types";
 import { set, get, remove, getDefaultValue, subscribe } from "./utils";
 import { StoreConfig } from "@latticexyz/store";
+import { curry } from "@latticexyz/common/utils";
 
 /**
  * Create a typed database client from a tuple database and a store config
  */
 export function createDatabaseClient<C extends StoreConfig>(database: TupleDatabase, config: C) {
   const _tupleDatabaseClient = new TupleDatabaseClient(database);
-  const client: Record<string, unknown> = { _tupleDatabaseClient };
+  const { namespace } = config;
+  const tables: Record<string, unknown> = {};
 
+  // Create utils with client argument prefilled
+  const utilsWithClient = {
+    set: curry(set<C>, _tupleDatabaseClient),
+    get: curry(get<C>, _tupleDatabaseClient),
+    remove: curry(remove<C>, _tupleDatabaseClient),
+    subscribe: curry(subscribe<C>, _tupleDatabaseClient),
+  };
+
+  // Create utils with client, namespace and table argument prefilled
   for (const table in config.tables) {
-    client[table] = {
+    tables[table] = {
       set: (key: Key<C, typeof table>, value: Value<C, typeof table>, options: SetOptions) =>
-        set(_tupleDatabaseClient, table, key, value, {
+        utilsWithClient.set(namespace, table, key, value, {
```

Comment:
afaik, tables can have a namespace override in config, so we may need to figure out how to handle that or forbid that e.g. one config per namespace

(preference for the former)

Reply:
I guess that brings up another question: should this use the resolved/expanded config, rather than the user defined one, because of things like this?

Imagining a fully resolved/expanded config would have a namespace defined per table (table namespace if defined, otherwise config-level namespace, otherwise empty), then we'd just use something like `table.namespace` here

Reply:
oh weird, it looks like the TS hints I was looking at were in a repo with broken TS, so my mud config was letting me set ~any key on the table (I had tried `namespace` to see if it was there)

<img width="532" alt="image" src="https://github.com/latticexyz/mud/assets/508855/e7f692b9-bee1-4545-9eba-a707079beaa2">